VmAgent.Core/ProcessOutputLogger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.Gaming.VmAgent.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using ApplicationInsights;
+    using ApplicationInsights.DataContracts;
+    using global::VmAgent.Core.Dependencies.Interfaces.Exceptions;
+    using Microsoft.Azure.Gaming.VmAgent.Core.Dependencies.Interfaces;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class ProcessOutputLogger: IProcessOutputLogger
+    {
+        private string _logFilePath;
+        private StreamWriter _logWriter;
+        private MultiLogger _logger;
+
+        public ProcessOutputLogger(string logFilePath, MultiLogger logger)
+        {
+            if (logFilePath == null)
+            {
+                throw new ProcessOuputLoggerCreationFailedException($"LogFilePath cannot be null.");
+            }
+
+            try
+            {
+                _logWriter = new StreamWriter(File.OpenWrite(logFilePath));
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation($"Exception was thrown while creating a Process Log file under {_logFilePath}. {ex}");
```

Comment:
you are logging _logFilePath before it is set with the right value, just remove the _ and you are good to go

Reply:
good catch! will fix that.